### PR TITLE
Add codemagic extension

### PR DESCRIPTION
**Pull request by Imgkl**
## Description

Features
- Trigger builds
  - Select the app you want to trigger a build for, Select the branch and the workflow and trigger the build.
  - If the build is in progress, you can cancel as well.
- See all builds
  - Get all the builds including the metadata.
  - Download the artifacts for a specific build.

## Screencast

https://github.com/user-attachments/assets/0f809eb7-1c33-4e9a-8539-5000b25e0e25

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by Imgkl** [comment]
The video attached is a bit outdated. I've added `useCachedPromise`.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- One of the commands crashes if I gives it wrong credentials, can you handle that gracefully?

codemagic 2024-10-10 at 12 00 13

- Remove metadata image 1 and 2 since they don't provide contextual value


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by Imgkl** [comment]
Hey @pernielsentikaer,

Thanks for taking your time and reviewing this PR.

> One of the commands crashes if I gives it wrong credentials, can you handle that gracefully?

When you say you gave wrong credentials? By that you mean invalid "Codemagic access Token"?

I've handled this scenario and can't reproduce the same error. 

https://github.com/user-attachments/assets/adb4970b-8da2-4220-95c4-f33556f779cb

**Comment by Imgkl** [comment]
Also @pernielsentikaer, I see the navigation action, "Open in PhpStorm". That's not something, I added in my code. That's related to your testing suite?

**Comment by pernielsentikaer** [comment]
As you can see below is one of the commands correctly handling wong credentials where the other doesn't, this is a fresh install - does that help you?

https://github.com/user-attachments/assets/a571b452-2184-4f39-acc4-4880da8612bc

**Comment by Imgkl** [comment]
I am working on this..

**Comment by Imgkl** [comment]
@pernielsentikaer Thanks for finding an edge case. This happens only on new install. Since I was using this extension for quite some time. It was hard for me to find this. You can take a look at the PR now. Thanks for your time.

File: extensions/codemagic/src/components/workflow-selector.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      await showToast(Toast.Style.Failure, "Failed to trigger build", String(error));
+    }
+  };
+
+  return (
+    <Form
+      navigationTitle={appName}
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title="Trigger Build" onSubmit={handleSubmit} />
```
Comment (pernielsentikaer) [inline extensions/codemagic/src/components/workflow-selector.tsx:40]: Missing an icon, all action panels should have an icon 🙂

File: extensions/codemagic/src/trigger-build.tsx [hunk 2]
```diff
@@ -143,16 +144,16 @@ const TriggerBuildCommand = () => {
...
                     />
+                    <Action title="Refresh Apps and Branches" onAction={revalidate} icon={Icon.Repeat} />
                     {app.lastBuild && cancellableStatuses.includes(app.lastBuild.status) && (
                       <Action
                         title="Cancel Build"
                         onAction={async () => {
                           await cancelBuild(app.lastBuild!._id);
                         }}
-                        icon="x-mark-circle-16"
+                        icon={{ source: Icon.XMarkCircle, tintColor: Color.Red }}
```
Comment (pernielsentikaer) [inline extensions/codemagic/src/trigger-build.tsx:154]: Here you can use ` style={Action.Style.Destructive}` intead of icon, that should make both an icon and style it 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**